src/api/app/controllers/webui/driver_update_controller.rb

```diff
@@ -98,7 +98,11 @@ def save
     services = @package.services
     services.removeService('generator_driver_update_disk')
     services.addService('generator_driver_update_disk', -1, dud_params)
-    services.save
+    unless services.save
+      flash.now[:error] = 'Error saving services file'
+      render :create
```

Comment:
No, because there are two points of return in this action, this is the second one, the first one is on line 70. So, just return with the error in the flash there. If everything was ok then do the default flash[:success] and redirection.